src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs

```diff
@@ -120,7 +120,7 @@ internal unsafe struct EnumerateMetafileProcMarshaller
             public EnumerateMetafileProcMarshaller(EnumerateMetafileProc? managed)
             {
                 _managed = managed is null ? null : (recordType, flags, dataSize, data, callbackData) =>
-                    managed(recordType, flags, dataSize, data, Marshal.GetDelegateForFunctionPointer<PlayRecordCallback>(callbackData)) ? Interop.BOOL.TRUE : Interop.BOOL.FALSE;
+                    managed(recordType, flags, dataSize, data, callbackData == IntPtr.Zero ? null! : Marshal.GetDelegateForFunctionPointer<PlayRecordCallback>(callbackData)) ? Interop.BOOL.TRUE : Interop.BOOL.FALSE;
```

Comment:
Rather than using `!` here, should EnumerateMetafileProc be declared to take a `PlayRecordCallback?` rather than a `PlayRecordCallback`?

Reply:
I think so, but that's a public API change and I wanted to make sure we got this fix in before .NET 7 ships as this is causing compatibility breaks.

Reply:
We should do both in 7 :)  If you don't want to do it here, can you please open a separate PR for it?  Thanks.

Reply:
Thanks.

Reply:
Opened https://github.com/dotnet/runtime/pull/71239